ocaml/xapi/locking_helpers.ml

```diff
@@ -223,3 +223,55 @@ module Named_mutex = struct
         finally f (fun () -> Thread_state.released r)
     )
 end
+
+module Semaphore = struct
+  type t = {
+      name: string
+    ; sem: Semaphore.Counting.t
```

Comment:
We are using `Xapi_stdext_threads.Semaphore` elsewhere. Can we replace all uses of that module with the standard library's module?

Reply:
The standard library's semaphore implementation is similarly with mutex+condvar, so yes. It doesn't have an 'execute' function though, and can only acquire 1 resource at a time (but I only see calls to acquire with the value of '1', so should be fine).
The one in stdext also has a spurius Condition.signal inside acquire that'll wake up threads for no good reason (it has taken away some resources so if they were sleeping before they still won't be able to acquire them).

Perhaps we could change Xapi_stdext.Semaphore to use the semaphore module as its implementation and act as a lightweight wrapper that provides the other convenience functions (execute), etc.

Reply:
We should change the implementation of the module with the stand library one